write-fonts/src/tables/glyf.rs

```diff
@@ -0,0 +1,298 @@
+            let point = iter.next()?;
+            let mut flag = SimpleGlyphFlags::empty();
+            let d_x = point.x - last_x;
+            let d_y = point.y - last_y;
+            last_x = point.x;
+            last_y = point.y;
+
+            if point.on_curve {
+                flag |= SimpleGlyphFlags::ON_CURVE_POINT;
+            }
+            let (x_flag, x_data) = flag_and_delta(
+                d_x,
+                SimpleGlyphFlags::X_SHORT_VECTOR,
+                SimpleGlyphFlags::X_IS_SAME_OR_POSITIVE_X_SHORT_VECTOR,
+            );
+            let (y_flag, y_data) = flag_and_delta(
+                d_y,
+                SimpleGlyphFlags::Y_SHORT_VECTOR,
+                SimpleGlyphFlags::Y_IS_SAME_OR_POSITIVE_Y_SHORT_VECTOR,
+            );
+
+            flag |= x_flag | y_flag;
+            Some((flag, x_data, y_data))
+        })
+    }
+}
+
+/// A little helper for managing how we're representing a given delta
+#[derive(Clone, Copy, Debug)]
+enum CoordDelta {
+    // this is a repeat (set in the flag) and so we write nothing
+    Skip,
+    Short(u8),
+    Long(i16),
+}
+
+impl FontWrite for CoordDelta {
+    fn write_into(&self, writer: &mut crate::TableWriter) {
+        match self {
+            CoordDelta::Skip => (),
+            CoordDelta::Short(val) => val.write_into(writer),
+            CoordDelta::Long(val) => val.write_into(writer),
+        }
+    }
+}
+
+impl<'a> IntoIterator for &'a Contour {
+    type Item = &'a CurvePoint;
+
+    type IntoIter = std::slice::Iter<'a, CurvePoint>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.0.iter()
+    }
+}
+
+impl FontWrite for SimpleGlyf {
+    fn write_into(&self, writer: &mut crate::TableWriter) {
+        assert!(self.contours.len() < i16::MAX as usize);
+        assert!(self._instructions.len() < u16::MAX as usize);
+        let n_contours = self.contours.len() as i16;
+        n_contours.write_into(writer);
+        let bbox = [self.x_max, self.y_max, self.x_min, self.y_min];
+        bbox.write_into(writer);
+        // now write end points of contours:
+        let mut cur = 0;
+        for contour in &self.contours {
+            cur += contour.len();
+            (cur as u16 - 1).write_into(writer);
+        }
+        (self._instructions.len() as u16).write_into(writer);
+        self._instructions.write_into(writer);
+
+        let deltas = self.compute_point_deltas().collect::<Vec<_>>();
+        //TODO: calculate flag repeats here
+        deltas
+            .iter()
+            .for_each(|(flag, _, _)| flag.bits().write_into(writer));
+        deltas.iter().for_each(|(_, x, _)| x.write_into(writer));
+        deltas.iter().for_each(|(_, _, y)| y.write_into(writer));
+    }
+}
+
+impl crate::validate::Validate for SimpleGlyf {
+    fn validate_impl(&self, _ctx: &mut crate::codegen_prelude::ValidationCtx) {
+        // pass
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use read::{FontData, FontRead};
+
+    use super::*;
+
+    #[test]
```

Comment:
Feels light on testing. 

Thinking "aloud," can we read a glyph from one of our binary test fonts, write it out, and get the same bytes?

Also, you have various error checks in the code, and no tests with of bad input.

Reply:
yes good points, see #219 :)